### ISSUE TITLE
Fresh install of conda, users can no longer create environments off a root install owned by root

### ISSUE THREAD

**Issue by Wildcarde** [conda/conda#3614]
This is a setup I've done at least 20 of at this point and it's generally worked just fine.
- Install as root into /opt/anaconda a variant of anaconda python (2.7 and 3.5 both failed).
- Grant users read access to this environment for use if they want to work from it. 
- Allow users to create sub environments if they want to make changes.  This last part is now quite broken and it seems to be related to assuming that users have write access to the central copy of anaconda:

I should state, I haven't done this in a month or two but this worked exactly as expected on the last setup and underpins several of our shared use machines where users can spin up an anaconda environment at will.

```
 conda create --name=psiturk python=2.7 anaconda pip
 Fetching package metadata .......
 Solving package specifications: ..........

 Package plan for installation in environment /home/kt11/.conda/envs/psiturk:

 The following NEW packages will be INSTALLED:

    _license:           1.1-py27_1          (soft-link)
    _nb_ext_conf:       0.3.0-py27_0        (soft-link)
...
    zeromq:             4.1.4-0             (soft-link)
    zlib:               1.2.8-3             (soft-link)

Proceed ([y]/n)? y

Linking packages ...
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
[jpeg                ]|###                                              WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                         |   0%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
[libgfortran         ]|#######                                                                                                                  WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
[openssl             ]|###########                                                                                                                                                                                      WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
[sqlite              ]|################           WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                               |   7%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
[zlib                ]|####################                                                                               WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]       |   9%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]
An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : linux-64
          conda version : 4.2.9
       conda is private : False
      conda-env version : 4.2.9
    conda-build version : 2.0.2
         python version : 2.7.12.final.0
       requests version : 2.11.1
       root environment : /opt/anaconda  (read only)
    default environment : /opt/anaconda
       envs directories : /home/<user>/.conda/envs
                          /opt/anaconda/envs
          package cache : /home/<user>/.conda/envs/.pkgs
                          /opt/anaconda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                          https://repo.continuum.io/pkgs/free/noarch/
                          https://repo.continuum.io/pkgs/pro/linux-64/
                          https://repo.continuum.io/pkgs/pro/noarch/
            config file : None
           offline mode : False



`$ /opt/anaconda/bin/conda create --name=psiturk python=2.7 anaconda pip`




    Traceback (most recent call last):
      File "/opt/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 473, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 144, in _main
        exit_code = args.func(args, p)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main_create.py", line 68, in execute
        install(args, parser, 'create')
      File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 405, in install
        execute_actions(actions, index, verbose=not context.quiet)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/plan.py", line 643, in execute_actions
        inst.execute_instructions(plan, index, verbose)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 134, in execute_instructions
        cmd(state, arg)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 78, in LINK_CMD
        link(state['prefix'], dist, lt, index=state['index'])
      File "/opt/anaconda/lib/python2.7/site-packages/conda/install.py", line 966, in link
        _link(src, dst, lt)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/install.py", line 171, in _link
        shutil.copy2(src, dst)
      File "/opt/anaconda/lib/python2.7/shutil.py", line 130, in copy2
        copyfile(src, dst)
      File "/opt/anaconda/lib/python2.7/shutil.py", line 82, in copyfile
        with open(src, 'rb') as fsrc:
    IOError: [Errno 13] Permission denied: u'/opt/anaconda/pkgs/dbus-1.10.10-0/libexec/dbus-daemon-launch-helper'

```

The anaconda 3 install didn't fail until the end:

```
[      COMPLETE      ]|################################################################################################################################################################################################################| 100%
Linking packages ...
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   0%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   0%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   0%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   1%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   1%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   2%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   2%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   3%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   3%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   4%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   4%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   5%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   5%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   6%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   6%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   7%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   7%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   8%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   8%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   9%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |   9%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |  10%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                                                                 |  10%
CondaOSError: OS error: failed to link (src='/opt/anaconda3/pkgs/dbus-1.10.10-0/libexec/dbus-daemon-launch-helper', dst='/home/kt11/.conda/envs/psiturk/libexec/dbus-daemon-launch-helper', type=3, error=PermissionError(13, 'Permission denied'))

```

We use this setup for a few different situations, in cases like this to get them a working environment they can't break while trying to make problematic code work, and in other situations to allow them to operate of a central copy of python that they can't change but can build off of if they need to.

**Comment by edurand** [comment]
We're having the same issue now.

**Comment by shwina** [comment]
Same issue to report.

**Comment by mforbes** [comment]
This is a problem for us too.

**Comment by vishnu2kmohan** [comment]
Running into the same issue.

**Comment by kadrlica** [comment]
I think this is related to #3469.

**Comment by AndrewPashkin** [comment]
Confirm with Anaconda-3 4.2.0.

**Comment by rcthomas** [comment]
Also here.  Are there just not tests to catch this kind of stuff, or are we all "wrong" for trying to do this...?

    conda info
    Current conda install:

                   platform : linux-64
              conda version : 4.2.9
           conda is private : False
          conda-env version : 4.2.9
        conda-build version : 1.21.1
             python version : 2.7.12.final.0
           requests version : 2.10.0
           root environment : /usr/common/usg/python/2.7-anaconda  (read only)
        default environment : /usr/common/usg/python/2.7-anaconda
           envs directories : /global/homes/r/rthomas/.conda/envs
                              /usr/common/usg/python/2.7-anaconda/envs
              package cache : /global/homes/r/rthomas/.conda/envs/.pkgs
                              /usr/common/usg/python/2.7-anaconda/pkgs
               channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                              https://repo.continuum.io/pkgs/free/noarch/
                              https://repo.continuum.io/pkgs/pro/linux-64/
                              https://repo.continuum.io/pkgs/pro/noarch/
                config file : /global/homes/r/rthomas/.condarc
               offline mode : False

**Comment by kalefranz** [comment]
> Are there just not tests to catch this kind of stuff

Actually, no, there aren't.  "Multi-user" hasn't ever officially been supported well in conda--at least not yet.  See issues going back years with "conda uninstalling itself."

We're fixing this in 4.3 and 4.4.  This PR going into 4.3: https://github.com/conda/conda/pull/3571
And then follow-ons for getting multi-user right in 4.4.

**Comment by rcthomas** [comment]
@kalefranz: Thanks for taking the time to outline the vision toward multi-user.  It is very much appreciated.

**Comment by AhmetTavli** [comment]
Same issue: conda create --name=tensorflow_env python=2.7

![conda_linking_error](https://cloud.githubusercontent.com/assets/8464220/19765300/3f87b0d6-9c50-11e6-99da-85d011d8c72a.jpg)

Any idea, how to solve this problem?

**Comment by TSchattschneider** [comment]
@AhmetTavli I had the same error when trying to create an environment for tensorflow. For me the reason was that I installed Anaconda using sudo. I had to remove it and reinstall normally without sudo, then that error didn't occur.

**Comment by AhmetTavli** [comment]
@TSchattschneider Thank you

**Comment by swt30** [comment]
We had this problem (permission error linking dbus). It was caused by permissions in the root conda install being too tight. Doing a `chmod -R a+r` on the folder makes all the files readable by everyone, which is what we needed.

**Comment by spinicist** [comment]
I've also hit this problem today on a multi-user install.

**Comment by spinicist** [comment]
Similarly to @swt30, our sysadmin tracked this issue down to conda placing the lock files in the root-owned anaconda/?.?/pkg directories. He changed the permissions on the pkg directory to 1777 as a quick fix - but it's not clear why anaconda is creating locks in the global directory when I'm trying to create a local environment in the first place?

**Comment by SwapnilParanjape** [comment]
I faced this problem today. I tried to run the command as root. Just did sudo conda create <<your command>> and that worked.

**Comment by fps697608** [comment]
Same issue here.
I've tried to create an environment on my two computers.
One was fine, the other one was unable.
Seems like that there are problems with a newer computer?

**Comment by Woodstonelee** [comment]
had similar issue here.  Now I'm using a workaround.  Find the repo of .tar.gz of the package you are trying to install. You can go to the channel URLs and search. Download the package directly to the .pkgs directory of conda in your home directory. The .pkgs could be located at ~/.conda/.pkgs or ~/anaconda_env/.pkgs. Also add the package download URL to the urls file inside the .pkgs folder. Then run conda install again, it will be installed successfully.

**Comment by jsm296** [comment]
There is at least one more packages than just the dbus package that have this odd permission problem, but it is an anomaly - I ran `ls -alR | grep "\-\-\-"` and it only found a handful of files (it only appeared in the world section of the permissions of the output).

In our particular install, we ran into this problem with dbus and one other package (the name of which I can't find right now).

This doesn't relate to the lock files, but is this problem of unreadable files in a few packages something that might just be an error in the creation of those packages?

**Comment by jsm296** [comment]
Also, making the files readable by world fixes the problem (as another commenter noted above).  These permissions also cause problems when doing other tasks in some circumstances - the unreadable files caused jupyter notebook server to not be able to run from the command line, as well, for example.

**Comment by kalefranz** [comment]
conda 4.3 is multi-user compatible and fixes these issues.

    conda update conda

If you're running 4.3 and you have a multi-user related problem, please file a new issue.

**Comment by samwisehawkins** [comment]
I still have the same issue, using conda 4.3

    ERROR conda.core.link:_execute_actions(337): An error occurred while installing package 'defaults::dbus-1.10.10-0'.
    IOError(13, 'Permission denied')
    Attempting to roll back.


    IOError(13, 'Permission denied')
